NETCORE/test/Microsoft.ApplicationInsights.AspNetCore.Tests/Extensions/ApplicationInsightsExtensionsTests/AddSingletonTests.cs

```diff
@@ -1,96 +0,0 @@
-﻿using System;
-using System.Collections.Generic;
-using System.ComponentModel;
-using System.Linq;
-using System.Text;
-
-using Microsoft.ApplicationInsights.Extensibility;
-using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
-using Microsoft.Extensions.DependencyInjection;
-using Microsoft.Extensions.DependencyInjection.Test;
-
-using Xunit;
-
-namespace Microsoft.ApplicationInsights.AspNetCore.Tests.Extensions.ApplicationInsightsExtensionsTests
-{
-    /// <summary>
-    /// When working with IServiceCossection, it can store three types of Implementations:
-    /// ImplementationFactory, ImplementationInstance, and ImplementationType.
-    /// We want to be able to add a Singleton but only if a user hasn't already done so.
-    /// This class is to test all the various edge cases.
-    /// </summary>
```

Comment:
The issue is as of 8.0.0 this summary is false.

There are new "keyed" forms of the possible registration types:

![image](https://github.com/user-attachments/assets/af6fe4fa-3156-43b4-b3fa-4bfe15902656)

`AddSingletonIfNotExists` could be made smarter to understand the new shape. But that would require an `8.0.0` bump for M.E.DI/.A in `Microsoft.ApplicationInsights.AspNetCore`.

Easier IMO to just call into `TryAddSingleton` and let it sort out what to do!